ta4j-core/src/main/java/org/ta4j/core/Trade.java

```diff
@@ -102,7 +102,8 @@ public Order getExit() {
     public boolean equals(Object obj) {
         if (obj instanceof Trade) {
             Trade t = (Trade) obj;
-            return entry.equals(t.getEntry()) && exit.equals(t.getExit());
+            return (entry == null ? t.getEntry() == null : entry.equals(t.getEntry()))
+            		&& (exit == null ? t.getExit() == null : exit.equals(t.getExit()));
         }
         return false;
     }
```

Comment:
Should this throw a `ClassCastException` if not `instanceof Trade`?  Also would be nice to see at least one new test case in the appropriate test class for any change to a pervasive core class.

Reply:
Test case would be appropriate. If obj is not instance of Trade the equals function returns false, that is standard behaviour of equals function

Reply:
equals-method-policy is defined in java.lang.Object#equals. I added some test-cases...